### ISSUE TITLE
Style/SafeNavigation does not handle the `.[]` method well

### ISSUE THREAD

**Issue by brandonweiss** [bbatsov/rubocop#4890]
The `Style/SafeNavigation` cop is maybe overly-strict in what it calls an offense. I think there is a high likelihood that in the case of the `.[]` method you might never want to make the change it's recommending.

A offense of `Style/SafeNavigation` can be easily corrected.

```ruby
foo && foo.bar

foo&.bar
```

But the `.[]` method can't be as-is.

```ruby
foo && foo[:bar]

foo&.[:bar] # Does not work
```

You could use `fetch` instead, although you have to be very careful to always add the `nil` so it doesn't raise.

```ruby
foo && foo[:bar]

foo&.fetch(:bar, nil)
```

However, not everything that looks like an Enumerable actually is.

```
foo = "bar".match(/(bar)/)
foo && foo[1]

foo = "bar".match(/(bar)/)
foo&.fetch(1) # Does not work, MatchData has no fetch method
```

--------

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-darwin16)
```

**Comment by Drenmi** [comment]
Fixed in #4749 and 0.51.1.

**Comment by brandonweiss** [comment]
@Drenmi Doh! Sorry! I forgot to search first. 🤦‍♂️

**Comment by Drenmi** [comment]
No problems. Can be hard to find, too, when closed, since the default search is for open issues. 🙂
